crates/nu-parser/src/parser.rs

```diff
@@ -3204,8 +3210,15 @@ pub fn parse_signature_helper(
 
                                         *default_value = Some(expression);
 
-                                        // Flags with a boolean type are just present/not-present switches
-                                        if var_type != &Type::Bool {
+                                        if expression_ty == Type::Bool {
+                                            error = error.or_else(|| {
+                                                Some(ParseError::AssignmentMismatch(
+                                                    "Default bool value".into(),
+                                                    format!("default value can't be bool"),
+                                                    expression_span,
+                                                ))
+                                            })
+                                        } else {
```

Comment:
Because defaulting to `false` would be confusing:

```
> def ready [--ready=false] { if $ready { 'Ready!' } else { 'Not ready!' } }
> ready --ready
Not ready!
```